### DISCUSSION TITLE
Custom handler with AWS Lambda HTTP extension

### DISCUSSION THREAD

**Discussion by michjea** [quarkusio/quarkus#40303] [Q&A]
I have a project with the AWS Lambda HTTP extension. My goal is to reuse the same codebase to deploy two lambdas, with different handlers. One with the AWS Lambda HTTP extension handler, and one with a custom handler.

With the AWS Lambda extension, we can define multiple handlers with `quarkus.lambda.handler=handler-name`. But when trying to do the same with `quarkus-amazon-lambda-http`, I've got a build error : 
`Caused by: io.quarkus.builder.BuildException: Build failure: Multiple handler classes.  You have a custom handler class and the AWS Lambda HTTP extension.  Please remove one of them from your deployment.`

So my question is why isn't that possible ?

[a comment by an automated account is not shown]

**Comment by lucas-gautier** [comment]
Found this via search - had a similar question when trying handle REST API Gateway events and Step Functions (SFN) events within one project. Trying to add a custom handler for SFN and REST gave the same error in the original post. A workaround is to use two separate Quarkus apps: 
- Step Functions only via custom `RequestHandler<SFNInput, SFNOutput>` via `quarkus-amazon-lambda`
- REST via API Gateway with stock Quarkus REST via `quarkus-amazon-lambda-rest`

This is not desirable because to the same code is needed just invoked in different ways. Moving common functionality to a jar is a lot of extra overhead resulting in 3 projects: 2 functions plus a shared jar. Looking at `LambdaHttpHandler.handleRequest` and before `nettyDispatch` is called might be where this handler could be modified to handle the request to another `@Named` bean when the `AwsProxyRequest` is not an API Gateway event.

Anyone else been able to get 1 Lambda function working via API Gateway and via Step Functions?

**Comment by tusharjajodia** [comment]
This is a much needed functionality for API + SQS setup as well. 

Amy one able to solve it ?
